### PR TITLE
feat: [CDS-69130]: Support for editing GitMetadata for pipeline and inputSet

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
### Summary

Changes are : 

- User will get an option to modify one or more of gitConnector, repo and filePath for remote pipelines and InputSets.
- To modify user can open the modalDialog from menu option Edit GitMetadata in the list page.
API responses handled handled : 
- If repo listing fails, user will see error in the dialog in right side similar to create/import experience.
- For edit API success/failure we are showing toaster.

#### Screenshots

Screenshot 2023-05-21 at 3 30 35 PM

Screenshot 2023-05-21 at 3 32 21 PM

Error experience
Screenshot 2023-05-23 at 11 26 49 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx [hunk 1]
```diff
@@ -963,6 +963,10 @@ function getConnectorSelectorSchema(selected: ConnectorSelectedValue, getString:
       return deafultSchema
   }
 }
+
+export const getConnectorValue = (connectorRef: ConnectorSelectedValue | string): string =>
```
Comment (sunnykesh-harness) [inline src/modules/35-connectors/components/ConnectorReferenceField/ConnectorReferenceField.tsx:967]: @pankajsingh-harness Added this utility as suggested.

File: src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx [hunk 2]
```diff
@@ -263,20 +265,22 @@ export function GitSyncForm<T extends GitSyncFormFields = GitSyncFormFields>(
...
-              // This is to handle auto fill after default selection, without it form validation will fail
-              if (formikProps.values.branch !== selected.value) {
-                formikProps.setFieldValue?.('branch', selected.value)
-              }
-            }}
-            selectedValue={defaultTo(formikProps.values.branch, branch)}
-            disabled={isEdit || disableFields.branch}
-            setErrorResponse={setErrorResponse}
-          />
+          {skipBranch ? null : (
```
Comment (sunnykesh-harness) [inline src/modules/40-gitsync/components/GitSyncForm/GitSyncForm.tsx:268]: Added `skipBranch` support to reuse GitSync form  for editGitMetadata use cases.

File: src/modules/70-pipeline/components/MigrateResource/EditGitMetadata.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    connectorRef: ''
+  },
+  resourceType,
+  identifier,
+  extraQueryParams,
+  onCancelClick,
+  onSuccess,
+  onFailure
+}: EditGitMetadataProps): JSX.Element {
+  const [errorResponse, setErrorResponse] = useState<ResponseMessage[]>()
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/MigrateResource/EditGitMetadata.tsx:53]: NIT: Being an array, we should use plurals for the namig.

**Comment by mayankVermaHarness** [comment]
A little late for feedback but for the error experience, should we show it on top of Modal instead of the right side?
We show/use ModalErrorHandler usually on the top of modals above all fields

**Review by pankajsingh-harness — APPROVED** [review]

**Comment by sunnykesh-harness** [comment]
> A little late for feedback but for the error experience, should we show it on top of Modal instead of the right side? We show/use ModalErrorHandler usually on the top of modals above all fields

@mayankVermaHarness This is as per [mock of similar modal ](https://www.figma.com/file/f6HSFuF1VceCtrPfMZWEIm/GitX2.0?type=design&node-id=602-19882&t=9uHhorAv1wg2fDMH-0) as git errors can be very big sometimes.

**Review by mayankVermaHarness — APPROVED** [review]

**Comment by sunnykesh-harness** [comment]
@YounixIO Have written enough tests for new files to get around 90% coverage, just branch coverage is below 80%. Please FM it.

File: src/modules/70-pipeline/components/MigrateResource/EditGitMetadata.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        break
+      case ResourceType.INPUT_SETS:
+        editInputSetGitMetadata(formValues)
+        break
+    }
+  }
+
+  const handleResponse = (response: ResponseMoveConfigResponse | ResponseInputSetMoveConfigResponseDTO): void => {
+    if (response.status === 'SUCCESS') {
+      setIsLoading(false)
```
[the remark this answers is not in the document]
Comment (sunnykesh-harness) [reply src/modules/70-pipeline/components/MigrateResource/EditGitMetadata.tsx:119]: Thanks for pointing out, even in success it is not required as modal itself is closed. Will clean up while adding same support for templates.

**Pull request merged**